tests/RPM/ParseSpec.hs

```diff
@@ -125,3 +127,87 @@ spec = describe "RPM.Parse" $ do
       stream ~?> parseSectionHeader `leavesUnconsumed` BS.pack []
       -- result is as expected
       stream ~> parseSectionHeader `parseSatisfies` (==expected)
+
+  describe "parseOneTag" $ do
+    it "returns Nothing when any of the input streams is empty" $ do
+      parseOneTag (BS.pack []) (BS.pack [])              `shouldBe` Nothing
+      parseOneTag (BS.pack [1, 2, 3, 4, 5]) (BS.pack []) `shouldBe` Nothing
+      parseOneTag (BS.pack []) (BS.pack [100, 8, 0, 7])  `shouldBe` Nothing
+
+    it "returns valid Tag when both input streams are valid" $ do
+      let store = BC.pack "123-test-me"
+      let bs = BS.pack [0, 0, 0, 100, -- tag
+                        0, 0, 0, 8, -- ty
+                        0, 0, 0, 4, -- offset
+                        0, 0, 0, 7 -- count
+                       ]
+      parseOneTag store bs `shouldBe` Just (HeaderI18NTable ["test-me"])
+
+  describe "parseSection" $
+    it "succeeds with valid data" $ do
+      let stream = BS.pack [
+            0x8E, 0xAD, 0xE8, -- section header signature
+            1, -- sectionVersion
+            0, 0, 0, 0, -- 4 reserved bytes
+            0, 0, 0, 7, -- sectionCount 4 bytes
+            0, 0, 0, 0xE4, -- sectionSize 4 bytes
+
+            -- tags defined in this section, sectionCount * 16 bytes
+            -- 62 7 212 16 == HeaderSignatures (Null), ty(7) /= 0, returns Nothing
+            0x00, 0x00, 0x00, 0x3e, 0x00, 0x00, 0x00, 0x07, 0x00, 0x00, 0x00, 0xd4, 0x00, 0x00, 0x00, 0x10,
```

Comment:
@clumens I was referring to this part of the header but then I figured it out. The ty values here do not match what the code expects so mkTag returns Nothing. I guess this is from an older version of RPM.